test/skimNtuple_HHbtag.cpp

```diff
@@ -1562,23 +1487,61 @@ int main (int argc, char** argv)
+									  theBigTree.daughters_pz->at(secondDaughterIndex),
+									  theBigTree.daughters_e->at(secondDaughterIndex));
+
+	  // electron energy scales and smears
+	  double uncorrEn1 = tlv_firstLepton.E();
+	  double uncorrEn2 = tlv_secondLepton.E();
+	  double corr1 = theBigTree.daughters_ecalTrkEnergyPostCorr->at(firstDaughterIndex);
+	  if(oph.isElectron(type1)) {
+		if (corr1 == -999.) {
+		  throw std::runtime_error("ERROR: unchanged electron energy scale!");
+		}
+		tlv_firstLepton *= corr1 / uncorrEn1;
+	  }
+	  double corr2 = theBigTree.daughters_ecalTrkEnergyPostCorr->at(secondDaughterIndex);
+	  if(oph.isElectron(type2)) {
+		if (corr2 == -999.) {
+		  throw std::runtime_error("ERROR: unchanged electron energy scale!");
+		}
+		tlv_secondLepton *= corr2 / uncorrEn2;
+	  }
+	  
+	  TLorentzVector tlv_firstLepton_eesUp    = tlv_firstLepton;
+	  TLorentzVector tlv_firstLepton_eesDown  = tlv_firstLepton;
+	  TLorentzVector tlv_secondLepton_eesUp   = tlv_secondLepton;
+	  TLorentzVector tlv_secondLepton_eesDown = tlv_secondLepton;
 
-	  const TLorentzVector tlv_secondLepton (theBigTree.daughters_px->at (secondDaughterIndex),
-											 theBigTree.daughters_py->at (secondDaughterIndex),
-											 theBigTree.daughters_pz->at (secondDaughterIndex),
-											 theBigTree.daughters_e->at (secondDaughterIndex)
-											 );
+	  TLorentzVector tlv_firstLepton_eerUp    = tlv_firstLepton;
+	  TLorentzVector tlv_firstLepton_eerDown  = tlv_firstLepton;
+	  TLorentzVector tlv_secondLepton_eerUp   = tlv_secondLepton;
+	  TLorentzVector tlv_secondLepton_eerDown = tlv_secondLepton;
 
-	  auto met_phi_corr = met_phi_correction_pxpy(
-												  theBigTree.METx->at(chosenTauPair),
+	  if (isMC) {
+		if(oph.isElectron(type1)) {
+		  tlv_firstLepton_eesUp   *= theBigTree.daughters_energyScaleUp  ->at(firstDaughterIndex) / uncorrEn1;
+		  tlv_firstLepton_eesDown *= theBigTree.daughters_energyScaleDown->at(firstDaughterIndex) / uncorrEn1;
+		  tlv_firstLepton_eerUp	  *= theBigTree.daughters_energySigmaUp  ->at(firstDaughterIndex) / uncorrEn1;				  
+		  tlv_firstLepton_eerDown *= theBigTree.daughters_energySigmaDown->at(firstDaughterIndex) / uncorrEn1;
+		}
+		if(oph.isElectron(type2)) {
+		  tlv_secondLepton_eesUp   *= theBigTree.daughters_energyScaleUp  ->at(secondDaughterIndex) / uncorrEn2;
+		  tlv_secondLepton_eesDown *= theBigTree.daughters_energyScaleDown->at(secondDaughterIndex) / uncorrEn2;
+		  tlv_secondLepton_eerUp   *= theBigTree.daughters_energySigmaUp  ->at(secondDaughterIndex) / uncorrEn2;				  
+		  tlv_secondLepton_eerDown *= theBigTree.daughters_energySigmaDown->at(secondDaughterIndex) / uncorrEn2;
+		}
+	  }
+	  
+	  auto met_phi_corr = met_phi_correction_pxpy(theBigTree.METx->at(chosenTauPair),
 												  theBigTree.METy->at(chosenTauPair),
-												  theBigTree.npv, theBigTree.RunNumber, PERIOD, isMC
-												  );
+												  theBigTree.npv, theBigTree.RunNumber, PERIOD, isMC);
```

Comment:
I am not sure what happened in this line...

Reply:
It's just the parenthesis; one line was removed. It does not affect the code.